packages/widget/src/components/network-selector/network-selector.ts

```diff
@@ -16,70 +17,91 @@ type Direction = (typeof Directions)[keyof typeof Directions];
+  selectedNetwork?: Domain;
 
-  @property({
-    attribute: false
-  })
+  @property({ attribute: false })
   onNetworkSelected?: (network?: Domain) => void;
 
-  @property({
-    type: Array
-  })
+  @property({ type: Array })
   networks: Domain[] = [];
 
-  onChange({ target }: Event): void {
-    const { value } = target as HTMLOptionElement;
-    const network = this.networks.find((n) => String(n.chainId) == value);
-    this.onNetworkSelected?.(network);
+  _toggleDropdown = (): void => {
+    this._isDropdownOpen = !this._isDropdownOpen;
+  };
+
+  _selectOption(option: Domain, event: Event): void {
+    event.stopPropagation();
+    this.selectedNetwork = option;
+    this._isDropdownOpen = false;
+    this.onNetworkSelected?.(option);
   }
 
-  renderEntries(): Generator<unknown, void> | HTMLTemplateResult {
-    if (this.networks) {
-      return map(this.networks, (entry: Domain) => {
-        // TODO: render network icon
-        return html`<option value=${entry.chainId} class="network-option">
-          ${capitalize(entry.name)}
-        </option>`;
-      });
-    }
-    return html`<option selected value="">Network</option>`;
+  _renderNetworkIcon(name: string): HTMLTemplateResult {
+    return networkIconsMap[name] || networkIconsMap.default;
+  }
+
+  _renderEntries(): Generator<unknown, void> | HTMLTemplateResult {
+    return map(
+      this.networks,
+      (network: Domain) => html`
+        <div
+          class="dropdownOption"
+          @click="${(e: Event) => this._selectOption(network, e)}"
+          role="option"
+        >
+          ${this._renderNetworkIcon(network.name)}
+          <span class="networkName">${capitalize(network.name)}</span>
+        </div>
+      `
+    );
+  }
+
+  _renderTriggerContent(): HTMLTemplateResult {
+    return this.selectedNetwork
+      ? html`${this._renderNetworkIcon(this.selectedNetwork.name)}
+          <span class="networkName">
+            ${capitalize(this.selectedNetwork.name)}
+          </span>`
+      : html`Select Network`;
   }
 
   render(): HTMLTemplateResult {
-    return html` <div class="selectorContainer">
+    return html`<div class="selectorContainer">
       <label for="selector" class="directionLabel">${this.direction}</label>
-      <section class="selectorSection">
-        <select
-          @change=${(event: Event) => this.onChange.bind(this)(event)}
-          ?disabled=${this.disabled}
-          class="selector"
+      <div
+        class="dropdown"
+        @click="${this._toggleDropdown}"
+        role="listbox"
+        tabindex="0"
+        aria-expanded="${this._isDropdownOpen ? 'true' : 'false'}"
+      >
+        <div class="dropdownTrigger">
+          <div class="selectedNetwork">${this._renderTriggerContent()}</div>
+          <div class="chevron ${this._isDropdownOpen ? 'open' : ''}">
+            ${chevronIcon}
+          </div>
+        </div>
+        <div
+          class="dropdownContent ${this._isDropdownOpen ? 'show' : ''}"
```

Comment:
I think the correct way to write this should be using `${}` and backticks, although Lit allows you to define this, but to me it looks kind of weird, since you are using template strings without backticks

Reply:
I think it works as we have top-level backticks from html``  ,so for such case, it's probably not required

Reply:
tottaly miss out that